tests/tests/test_state_scripts.py

```diff
@@ -709,16 +704,17 @@ def test_state_scripts(self, test_set):
             # quite slow.
             subprocess.check_call(["tar", "czf", "../rootfs-scripts.tar.gz", "."], cwd=rootfs_script_dir)
             # Stop client first to avoid race conditions.
-            run("systemctl stop mender")
+            execute(run, "systemctl stop mender", hosts=[client])
```

Comment:
I'm not exactly sure why, but if you recurse into `execute()`, sometimes, but not always, you lose all the exception info if the error happens in there. By calling it individually where we need it, we keep this info in all places we are not running `execute()`, which is most of the code.